extensions/amp-story/0.1/amp-story-page.js

```diff
@@ -575,6 +612,66 @@ export class AmpStoryPage extends AMP.BaseElement {
           EventType.DEV_LOG_ENTRIES_AVAILABLE, logEntries, {bubbles: true});
     });
   }
+
+
+  /**
+   * Displays a loading spinner whenever the video is buffering.
+   * Has to be called after the mediaPool preload method, that swaps the video
+   * elements with new amp elements.
+   * @private
+   */
+  startListeningToVideoEvents_() {
+    const videos = this.getAllVideos_();
+
+    if (videos.length === 0) {
+      return;
+    }
+
+    this.debounceToggleLoadingSpinner_(true);
+    videos.forEach(videoEl => {
+      this.unlisteners_.push(listen(
+          videoEl, 'playing', () => this.debounceToggleLoadingSpinner_(false)));
+      this.unlisteners_.push(listen(
+          videoEl, 'waiting', () => this.debounceToggleLoadingSpinner_(true)));
+    });
+  }
+
+
+  /**
+   * @private
+   */
+  stopListeningToVideoEvents_() {
+    this.unlisteners_.forEach(unlisten => unlisten());
+    this.unlisteners_.length = 0;
```

Comment:
This seems odd... why do we need to set this to 0?

Reply:
It's a trick to reset the array without creating a new array. But you're right it's confusing and doesn't make any difference, I'm replacing it with ``this.unlisteners_ = [];``